src/api/shopping.py

```diff
@@ -63,13 +55,6 @@ def optimize_shopping_route(user_id: int, food_id: int, use_preferences: bool):
                     detail=f"User does not exist, {e}"
                 )
 
-            # If using preferences, check if preferences exist
-            if use_preferences and user_info.budget is None:
-                raise HTTPException(
-                    status_code=status.HTTP_404_NOT_FOUND,
-                    detail="User preferences not set"
-                )
-                
             try:
                 result = conn.execute(find_matching_store_ids_query, food_data)
             except NoResultFound as e:
```

Comment:
This except will never trigger, even if an invalid food id is requested. I believe NoResultFound only triggers on statements with .one() or .scalar_one() that return no results, like the previous try-except statement.

Reply:
This is false. You can test it in the api, both exceptions can and will occur.

Reply:
What input did you use to get the response "Food item does not exist"?

Reply:
Nevermind was kinda confused by the comment review left. Specifically the code that was highlighted. Yea that exception shouldn't ever run so ill remove it

Reply:
Ok now I'm confused. When I run it with food id -1 without that exception it gives internal server error. Honestly I don't even know anymore at this point, but I know it needs to be in the code or else internal server error

Reply:
nevermind i was being stupid again